spark/v3.1/spark/src/main/scala/org/apache/spark/sql/execution/datasources/SparkExpressionConverter.scala

```diff
@@ -30,4 +33,18 @@ object SparkExpressionConverter {
     // But these two conversions already exist and well tested. So, we are going with this approach.
     SparkFilters.convert(DataSourceStrategy.translateFilter(sparkExpression, supportNestedPredicatePushdown = true).get)
   }
+
+  @throws[AnalysisException]
+  def collectResolvedSparkExpression(session: SparkSession, tableName: String, where: String): Expression = {
+    var expression: Expression = null
+    // Add a dummy prefix linking to the table to collect the resolved spark expression from optimized plan.
+    val prefix = String.format("SELECT 42 from %s where ", tableName)
+    val logicalPlan = session.sessionState.sqlParser.parsePlan(prefix + where)
+    val optimizedLogicalPlan = session.sessionState.executePlan(logicalPlan).optimizedPlan
```

Comment:
The only change compared to spark-3.2 PR is this. `executePlan` accepts only one argument in spark-3.0 and spark-3.1